test/emqx_shared_sub_SUITE.erl

```diff
@@ -37,11 +37,15 @@ all() -> emqx_ct:all(?SUITE).
 init_per_suite(Config) ->
     net_kernel:start(['master@127.0.0.1', longnames]),
     emqx_ct_helpers:boot_modules(all),
+    PortDiscovery = application:get_env(gen_rpc, port_discovery),
+    application:set_env(gen_rpc, port_discovery, stateless),
+    application:ensure_all_started(gen_rpc),
     emqx_ct_helpers:start_apps([]),
-    Config.
+    [{port_discovery, PortDiscovery} | Config].
 
-end_per_suite(_Config) ->
-    emqx_ct_helpers:stop_apps([]).
+end_per_suite(Config) ->
+    emqx_ct_helpers:stop_apps([gen_rpc]),
+    application:get_env(gen_rpc, port_discovery, proplists:get_value(port_discovery, Config)).
```

Comment:
Shouldn't this be `application:put_env`?

Also, the result of `get_env` in `init_suite` might be `{ok, Value} | undefined`, so it should be checked before restored.

Reply:
Nice catch